presto-native-execution/src/test/java/com/facebook/presto/nativeworker/PrestoNativeQueryRunnerUtils.java

```diff
@@ -401,6 +405,105 @@ public QueryRunner build()
     }
 
+    public static DeltaQueryRunnerBuilder nativeDeltaQueryRunnerBuilder()
+    {
+        return new DeltaQueryRunnerBuilder(QueryRunnerType.NATIVE);
+    }
+
+    public static DeltaQueryRunnerBuilder javaDeltaQueryRunnerBuilder()
+    {
+        return new DeltaQueryRunnerBuilder(QueryRunnerType.JAVA);
+    }
+
+    public static class DeltaQueryRunnerBuilder
+    {
+        private NativeQueryRunnerParameters nativeQueryRunnerParameters = getNativeQueryRunnerParameters();
+        private Path dataDirectory = nativeQueryRunnerParameters.dataDirectory;
+        private String serverBinary = nativeQueryRunnerParameters.serverBinary.toString();
+        private Integer workerCount = nativeQueryRunnerParameters.workerCount.orElse(4);
+        private CatalogType catalogType = Optional
+                .ofNullable(nativeQueryRunnerParameters.runnerParameters.get("delta.catalog.type"))
+                .map(v -> CatalogType.valueOf(v.toUpperCase()))
+                .orElse(CatalogType.HIVE);
+        private Integer cacheMaxSize = 0;
+        private Map<String, String> extraProperties = new HashMap<>();
+        private Map<String, String> extraConnectorProperties = new HashMap<>();
+        private Optional<String> remoteFunctionServerUds = Optional.empty();
+        private TimeZoneKey timeZoneKey = TimeZoneKey.getTimeZoneKey(TimeZone.getDefault().getID());
+        private boolean caseSensitiveParitions;
+        // External worker launcher is applicable only for the native iceberg query runner, since it depends on other
+        // properties it should be created once all the other query runner configs are set. This variable indicates
+        // whether the query runner returned by builder should use an external worker launcher, it will be true only
+        // for the native query runner and should NOT be explicitly configured by users.
+        private boolean useExternalWorkerLauncher;
+
+        private DeltaQueryRunnerBuilder(QueryRunnerType queryRunnerType)
+        {
+            if (queryRunnerType.equals(QueryRunnerType.NATIVE)) {
+                this.extraProperties.putAll(ImmutableMap.<String, String>builder()
+                        .put("http-server.http.port", "8080")
+                        .put("query.max-stage-count", "110")
+                        .putAll(getNativeWorkerSystemProperties())
+                        .build());
+                this.useExternalWorkerLauncher = true;
+            }
+            else {
+                this.extraProperties.putAll(ImmutableMap.of(
+                        "regex-library", "RE2J",
+                        "offset-clause-enabled", "true",
+                        "query.max-stage-count", "110"));
+                this.extraConnectorProperties.putAll(ImmutableMap.of("hive.parquet.writer.version", "PARQUET_1_0"));
+                this.useExternalWorkerLauncher = false;
+            }
+        }
+
+        public DeltaQueryRunnerBuilder setUseThrift(boolean useThrift)
```

Comment:
Seems like we only invoke these functions from the build function below and don't really intend to call these from outside. Should these be made private then ? Or else we can remove them entirely as well.